src/backend.rs

```diff
@@ -26,6 +27,7 @@ impl From<Config> for Backend {
         Self {
             path: config.path,
             output_dir: config.output_dir,
+            source_dir: PathBuf::from("src"),
```

Comment:
i don't love this approach. You set the source directory to "src" but immediately overwrite it again anyway.

it might be better to remove the
```rust
#[derive(Deserialize)]
#[serde(from = "Config")]
```

from the `Backend` struct, introduce a `Backend::new(config: Config, source_dir: PathBuf)` method and then call that from `from_context`

something like:

```rust
    pub fn new(config: Config, source_dir: PathBuf) -> Self {
        Self {
            path: config.path,
            output_dir: config.output_dir,
            inline: config.inline,
            layout: config.layout,
            source_dir,
        }
    }

    pub fn from_context(ctx: &PreprocessorContext) -> Self {
        let config: Config = ctx.config.get_preprocessor("d2")
            .expect("d2 preprocessor config not found")
            .clone()
            .try_into()
            .expect("Failed to parse d2 config");

        Self::new(config, ctx.config.book.src.clone())
    }
```